ironfish/src/network/peers/connections/webSocketConnection.ts

```diff
@@ -17,24 +18,26 @@ import { NetworkError } from './errors'
 export class WebSocketConnection extends Connection {
   private readonly socket: IsomorphicWebSocket
 
-  // The hostname of the address that was used to establish the WebSocket connection, if any
-  readonly hostname?: string
+  private _address: WebSocketAddress | null
+  get address(): WebSocketAddress | null {
+    return this._address
+  }
 
-  // The port of the address that was used to establish the WebSocket connection, if any
-  port?: number
+  set port(port: number) {
+    this._address = this._address ? { host: this._address.host, port } : null
```

Comment:
What do you think about a more traditional if check? I don't think it's necessary to create a new object just to set the port

```suggestion
    if (this._address) {
      this._address.port = port
    }
```

Reply:
yea sounds good to me